pkg/controller/machine/machine_controller.go

```diff
@@ -274,15 +278,33 @@ func (r *ReconcileMachine) isDeleteAllowed(machine *clusterv1.Machine) bool {
 	return node.UID != machine.Status.NodeRef.UID
 }
 
-func (r *ReconcileMachine) deleteNode(ctx context.Context, name string) error {
-	var node corev1.Node
-	if err := r.Client.Get(ctx, client.ObjectKey{Name: name}, &node); err != nil {
-		if apierrors.IsNotFound(err) {
-			klog.V(2).Infof("Node %q not found", name)
-			return nil
+func (r *ReconcileMachine) deleteNode(ctx context.Context, cluster *clusterv1.Cluster, name string) error {
+	if cluster == nil {
```

Comment:
ah this was a little bit confusing to read through. could consider a comment here because a nil cluster clearly means something special is happening, but I don't know why a nil cluster implies the management cluster is the workload cluster

Reply:
Added some comments

Reply:
Added some comments, in case Cluster is specified, we look for the KubeConfig secret